{"version":3,"file":"errorHandler.js","sources":["../../../packages/utils/errorHandler.ts"],"sourcesContent":["import { nextTick } from \"vue\";\nimport { consoleError, consoleWarn } from \"@fast-element-plus/utils\";\nimport { AxiosError } from \"axios\";\nimport { ElNotification } from \"element-plus\";\n\n/**\n * 全局异常捕获\n */\nexport const errorHandler = (error: any): boolean => {\n\tif (!error) return;\n\t// 过滤 Axios 错误\n\tif (error instanceof AxiosError) {\n\t\treturn false;\n\t}\n\tconst errorMap: any = {\n\t\tInternalError: \"Javascript引擎内部错误\",\n\t\tReferenceError: \"未找到对象\",\n\t\tTypeError: \"使用了错误的类型或对象\",\n\t\tRangeError: \"使用内置对象时，参数超范围\",\n\t\tSyntaxError: \"语法错误\",\n\t\tEvalError: \"错误的使用了Eval\",\n\t\tURIError: \"URI错误\",\n\t\tAggregateError: \"未知的多个错误\",\n\t\tTimeoutError: \"操作超时\",\n\t\tNetworkError: \"网络错误\",\n\t\tOutOfMemoryError: \"内存溢出\",\n\t};\n\tif (error === \"cancel\") {\n\t\tconsoleWarn(\"Cancel\", \"操作已取消\");\n\t} else {\n\t\tconst errorName = errorMap[error.name] || \"未知错误\";\n\t\tconsoleError(\"Handler\", error);\n\t\tnextTick(() => {\n\t\t\tElNotification({\n\t\t\t\ttitle: \"系统错误\",\n\t\t\t\tmessage: errorName,\n\t\t\t\tduration: 3000,\n\t\t\t\tposition: \"top-right\",\n\t\t\t});\n\t\t});\n\t}\n};\n"],"names":["error","AxiosError","errorMap","InternalError","ReferenceError","TypeError","RangeError","SyntaxError","EvalError","URIError","AggregateError","TimeoutError","NetworkError","OutOfMemoryError","errorName","name","nextTick","ElNotification","title","message","duration","position"],"mappings":"+NAQ6BA,IAC5B,IAAKA,EAAO,OAER,GAAAA,aAAiBC,EAAAA,WACb,OAAA,EAER,MAAMC,EAAgB,CACrBC,cAAe,mBACfC,eAAgB,QAChBC,UAAW,cACXC,WAAY,gBACZC,YAAa,OACbC,UAAW,aACXC,SAAU,QACVC,eAAgB,UAChBC,aAAc,OACdC,aAAc,OACdC,iBAAkB,QAEnB,GAAc,WAAVb,gBACS,SAAU,aAChB,CACN,MAAMc,EAAYZ,EAASF,EAAMe,OAAS,sBAC7B,UAAWf,GACxBgB,EAAAA,UAAS,KACOC,iBAAA,CACdC,MAAO,OACPC,QAASL,EACTM,SAAU,IACVC,SAAU,aACV,GAEH"}